tests/broadcaster_tests/test_broadcaster.py

```diff
@@ -45,22 +48,10 @@ def test_update_data(self):
     @patch('src.broadcaster.broadcaster.pub')
     def test_read_data(self, mock_open, mock_pub):
         """ Tests read data """
-        self.assertEqual(self.testable.read_data(), None)
-        self.assertEqual(self.messenger.read_data(), None)
-        self.assertEqual(self.filewriter.read_data(), None)
-
-        self.testable.update_data(self.data)
-        self.messenger.update_data(self.data)
-        self.filewriter.update_data(self.data)
-
-        self.assertEqual(self.testable.read_data(), None)
-        self.assertEqual(self.messenger.read_data(), None)
-        self.assertEqual(self.filewriter.read_data(), None)
-
-        self.assertEqual(self.testable.read_data(key="inval"), None)
-        self.assertEqual(self.messenger.read_data(key="inval"), None)
-        self.assertEqual(self.filewriter.read_data(key="inval"), None)
+        self.testable.publish_dictionary(self.data)
+        self.messenger.publish_dictionary(self.data)
+        self.filewriter.publish_dictionary(self.data)
 
-        self.assertEqual(self.testable.read_data(key=1), "test")
-        self.assertEqual(self.messenger.read_data(key=2), "test2")
-        self.assertEqual(self.filewriter.read_data(key=3), "test3")
+        self.assertEqual(self.testable.publish_key(key=1), "test")
+        self.assertEqual(self.messenger.publish_key(key=2), "test2")
```

Comment:
rather than check if it returns value correctly, check that it publishes correctly to pubsub